j2v/generation/generator.py

```diff
@@ -58,10 +58,13 @@ def collect_all_paths(self, data_object, current_path=None, current_view=None, d
                                        data_object_key=key,
                                        parent_object_key=data_object_key)
         elif is_non_empty_1D_list(data_object):
-            view_name = self.create_view_name(current_view, current_path)
-            sample_element = data_object[0]
+            view_name = self.create_view_name(current_view, current_path)            
             self.add_explore_join(view_name, current_view, current_path)
-            self.collect_all_paths(data_object=sample_element,
+            # Sample size i.e. the number of elements that will be sampled/processed inside an array in the JSON
+            array_sample_size = 50            
+            data_objects_selected = data_object if len(data_object) <= array_sample_size else random.sample(data_object, array_sample_size)
```

Comment:
Could be done nicer:
`data_objects_selected  = random.sample(data_object,min(len(data_object), array_sample_size))`